### PR TITLE
Left Join-ing on roles to allow users without roles to login

### PR DESCRIPTION
**Pull request by johnhuffsmith**
Looks like a change in https://github.com/directus/directus/pull/10663 added an `inner join` on `directus_roles` on the user login lookup query with the unintended side effect that users without roles can no longer login. See: https://github.com/directus/directus/pull/10663/files#diff-03b58901729f938551b47a9357d79ae1d0c62868265a05e8f9980d30b22c0e95R77

This changes that line to a `left join` so the role fields will return as `null`

Adds a test for this condition to prevent future regressions.

*Bonus; adds a `filter` to the migration objects list witch solves an edge case where `.map` files were getting included as migration candidates (after enabling sourceMap) making it look like there are duplicate migration keys and throwing the duplicate key check exception.

### CODE REVIEW

File: api/src/database/migrations/run.ts [hunk 1]
```diff
@@ -24,7 +24,9 @@ export default async function run(database: Knex, direction: 'up' | 'down' | 'la
 	const migrations = [
 		...migrationFiles.map((path) => parseFilePath(path)),
 		...customMigrationFiles.map((path) => parseFilePath(path, true)),
-	].sort((a, b) => (a.version > b.version ? 1 : -1));
+	]
+		.sort((a, b) => (a.version > b.version ? 1 : -1))
+		.filter((m) => m.file.endsWith('.js'));
```
Comment (azrikahar) [inline api/src/database/migrations/run.ts:29]: We should be able to move this to line 18 since there's existing filtering logic there, with customMigrations' filtering in line 20.

**Review by rijkvanzanten — APPROVED** [review]
